src/createServer.js

```diff
@@ -45,6 +45,7 @@ function createServer (options = {}) {
   server.onlineModeExceptions = Object.create(null)
   server.favicon = favicon
   server.options = options
+  options.registryCodec = options.registryCodec || mcData.registryCodec || mcData.loginPacket?.dimensionCodec
```

Comment:
The dimensionCodec is no longer in loginPacket. So it it needs to be dumped and put into mcdata seperately. It's required before clients are able to join (to exit configuration state)